server/core/storage_v2.go

```diff
@@ -0,0 +1,623 @@
+// Copyright 2021 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"encoding/json"
+	"math"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/encryption"
+	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/server/encryptionkm"
+	"github.com/tikv/pd/server/kv"
+	"go.etcd.io/etcd/clientv3"
+)
+
+// StorageV2 is the interface for the backend storage of the PD.
+// V2 is just a temp name, we will rename it to Storage eventually.
+// TODO: replace the core.Storage with this interface later.
+type StorageV2 interface {
+	ConfigStorage
+	MetaStorage
+	RuleStorage
+	ComponentStorage
+	ReplicationStatusStorage
+	GCSafePointStorage
```

Comment:
It's a little bit specific.

Reply:
It contains some specific `GCSafePoint` handling logic, it's hard to make it a generic definition I think.